airflow/operators/trigger_dagrun.py

```diff
@@ -70,13 +71,24 @@ class TriggerDagRunLink(BaseOperatorLink):
     name = "Triggered DAG"
 
     def get_link(self, operator: BaseOperator, *, ti_key: TaskInstanceKey) -> str:
-        # Fetch the correct dag_run_id for the triggerED dag which is
-        # stored in xcom during execution of the triggerING task.
+        # Fetch the correct dag id and run id for the triggerED dag
+        # which is stored in xcom during execution of the triggerING task.
+        dag_id = XCom.get_value(ti_key=ti_key, key=XCOM_DAG_ID)
         triggered_dag_run_id = XCom.get_value(ti_key=ti_key, key=XCOM_RUN_ID)
-        query = {
-            "dag_id": cast(TriggerDagRunOperator, operator).trigger_dag_id,
-            "dag_run_id": triggered_dag_run_id,
-        }
+
+        old_trigger_dag_id: str | None = str(cast(TriggerDagRunOperator, operator).trigger_dag_id)
+        if (
+            isinstance(old_trigger_dag_id, str)
+            and old_trigger_dag_id.startswith("{{")
+            and old_trigger_dag_id.endswith("}}")
+        ):
+            old_trigger_dag_id = None
```

Comment:
Is this check enough? Like, can the template not be something like `my_dag_{{ ti.task_id }}` which would make this check fail?

Reply:
thanks, I've updated this check.